### PR TITLE
refactor, instead of using `is721` flag, use `type` flag instead. Add `BeforeAfterBalance` to swap identification

### PR DESCRIPTION
**Pull request by KillariDev**
this should fix https://github.com/DarkFlorist/TheInterceptor/issues/317 as well. But I wasn't able to generate such transaction with uniswap now to test it.

### CODE REVIEW

File: extension/app/ts/background/background.ts [hunk 1]
```diff
@@ -247,6 +245,8 @@ async function handleSigningMode(
 	try {
 		parsedRequest = EthereumJsonRpcRequest.parse(request.options)
 	} catch (error) {
+		console.log(request)
+		console.log(error)
```
Comment (MicahZoltu) [inline extension/app/ts/background/background.ts:249]:
```suggestion
		console.error(request)
		console.error(error)
```

File: extension/app/ts/background/background.ts [hunk 2]
```diff
@@ -162,6 +158,8 @@ async function handleSimulationMode(
...
```
Comment (MicahZoltu) [inline extension/app/ts/background/background.ts:162]:
```suggestion
		console.error(request)
		console.error(error)
```

**Review by MicahZoltu — APPROVED** [review]

File: extension/app/ts/background/background.ts (hunk 1 above)
Comment (MicahZoltu) [reply extension/app/ts/background/background.ts:249]: What do you mean?  I don't think an app has insight into console errors.  The only thing this should do is change the color and how it gets filtered when using dev tooling.

Comment (KillariDev) [reply extension/app/ts/background/background.ts:249]: Chrome shows them as errors for the extension

Comment (KillariDev) [reply extension/app/ts/background/background.ts:249]: image
this button appears

**Pull request merged**